### PR TITLE
[stable/ghost] release 0.4.13

### PR DESCRIPTION
**Pull request by sameersbn**
- Fixes issue reported in https://github.com/bitnami/bitnami-docker-ghost/issues/56 /cc @luxas

### CODE REVIEW

File: stable/ghost/templates/NOTES.txt
Comment (luxas) [inline stable/ghost/templates/NOTES.txt]: I wouldn't use `kubectl port-forward` for this, which talks to one pod only
Instead I'd use the service that does all this for us:
```
export SVC_IP=$(kubectl get service --namespace {{ .Release.Namespace }} {{ template "fullname" . }} -o jsonpath="{.spec.clusterIP}")
echo Blog URL:  http://${SVC_IP}{{ if .Values.ghostPort }}:{{ .Values.ghostPort  }}{{ end }}
echo Admin URL: http://${SVC_IP}{{ if .Values.ghostPort }}:{{ .Values.ghostPort  }}{{ end }}/ghost
```

**Review by luxas — CHANGES_REQUESTED** [review]
You have to update `ports[0].port` as well, should be `{{ default "80" .Values.ghostPort }}`

But the reason I filed https://github.com/bitnami/bitnami-docker-ghost/issues/56 was that even though I manually changed `GHOST_PORT_NUMBER`, it still just listened on `2368`.

So I think there's a bug in the image as well.

File: stable/ghost/templates/NOTES.txt
Comment (prydonius) [reply stable/ghost/templates/NOTES.txt]: That service IP likely isn't accessible from outside the cluster though? I'm open to changing the way we describe how ClusterIP services can be accessed, but I would argue that this isn't a big issue given that the default service type is LoadBalancer (so you'd explicitly be choosing to use ClusterIP for some reason) and Ghost only runs in a single pod anyway.

**Comment by sameersbn** [comment]
> even though I manually changed GHOST_PORT_NUMBER, it still just listened on 2368.

That is by design. `GHOST_PORT_NUMBER` is not supposed to change the port on which the ghost app server listens on. It's meant for the generation of the correct URL's in the ghost application.

For example, if you have a ghost instance running on `2368` and an nginx proxy listening on port 80, you would have to set `GHOST_PORT_NUMBER=80`. This is exactly the issue handled by the first commit and that it the ghost kubernetes service did not respect the value of `ghostPort`

File: stable/ghost/templates/NOTES.txt
Comment (sameersbn) [reply stable/ghost/templates/NOTES.txt]: Yes, as @prydonius pointed out, accessing the service using the cluster IP will probably only work with solutions such as minikube.

**Review by luxas — APPROVED** [review]
> That is by design. GHOST_PORT_NUMBER is not supposed to change the port on which the ghost app server listens on. It's meant for the generation of the correct URL's in the ghost application.

Ah, I see. Makes sense.

https://github.com/bitnami/bitnami-docker-ghost#environment-variables
> GHOST_PORT_NUMBER: Ghost application port. Default: 80

I think the documentation could be updated on that point then. I always interpreted this as "the port to listen on" vs "the port to advertise"

> That service IP likely isn't accessible from outside the cluster though? I'm open to changing the way we describe how ClusterIP services can be accessed, but I would argue that this isn't a big issue given that the default service type is LoadBalancer (so you'd explicitly be choosing to use ClusterIP for some reason) and Ghost only runs in a single pod anyway.

Ok, we can keep this as-is if you like then. I think that using ClusterIP is probably pretty common, as you might want to use an Ingress controller in front of your Service (regardless of if you are able to use type=LoadBalancer or not)

LGTM :+1:

**Comment by sameersbn** [comment]
> GHOST_PORT_NUMBER: Ghost application port. Default: 80

That's a misleading statement. We should fix that.

> you might want to use an Ingress controller in front of your Service

Yes, that's the logical next step for the chart.

Thank for taking the time to review the changes.

**Comment by luxas** [comment]
@prydonius looks good to you as well? Then I think we can merge this
How often are the GCS buckets updated? On a periodic basis or triggered by merges?

**Review by prydonius — APPROVED** [review]
lgtm! @luxas it should be updated after merge (give a couple minutes for cache to invalidate)

**Pull request merged**
